The first Travis OSX error is
```
/Users/travis/build/JuliaLang/julia/src/codegen.cpp:1:1: error: unknown type name 'wnloadQueue'
wnloadQueue</string>
```
Maybe the codegen.cpp file got corrupted somehow? Also, for the other errors, the reported line numbers seem to be off, so maybe the beginning of that file was garbled? I'll restart the build, let's see whether that helps...

--- a/base/essentials.jl
+++ b/base/essentials.jl
@@ -369,6 +369,6 @@ call obsolete versions of a function `f`.
 """
 function invokelatest(f, args...; kwargs...)
     # We use a closure (`inner`) to handle kwargs.
-    inner(f, args...) = f(args...; kwargs...)
-    Core._apply_latest(inner, (f, args...))
+    inner() = f(args...; kwargs...)
+    Core._apply_latest(inner)
 end

--- a/test/misc.jl
+++ b/test/misc.jl
@@ -722,19 +722,40 @@ if Bool(parse(Int,(get(ENV, "JULIA_TESTFULL", "0"))))
     end
 end
 
-# invokelatest function for issue #19774
-issue19774(x) = 1
-let foo() = begin
-        eval(:(issue19774(x::Int) = 2))
-        return Base.invokelatest(issue19774, 0)
-    end
-    @test foo() == 2
+# Test issue #19774 invokelatest fix.
+
+# we define this in a module to allow rewriting
+# rather than needing an extra eval.
+module Issue19774
+f(x) = 1
 end
 
-kwargs19774(x, y; z=0) = x * y + z
+# First test the world issue condition.
 let foo() = begin
-        eval(:(kwargs19774(x::Int, y::Int; z=3) = z))
-        return Base.invokelatest(kwargs19774, 2, 3; z=1)
+        Issue19774.f(x::Int) = 2
+        return Issue19774.f(0)
+    end
+    @test foo() == 1    # We should be using the original function.
+end
+
+# Now check that invokelatest fixes that issue.
+let foo() = begin
+        Issue19774.f(x::Int) = 3
+        return Base.invokelatest(Issue19774.f, 0)
+    end
+    @test foo() == 3
+end
+
+# Check that the kwargs conditions also works
+module Kwargs19774
+f(x, y; z=0) = x * y + z
+end
+
+@test Kwargs19774.f(2, 3; z=1) == 7
+
+let foo() = begin
+        Kwargs19774.f(x::Int, y::Int; z=3) = z
+        return Base.invokelatest(Kwargs19774.f, 2, 3; z=1)
     end
     @test foo() == 1
 end

